src/plugins/media/media.js

```diff
@@ -36,60 +59,107 @@
         onStepleave,
         onPlay,
         onPause,
-        onEnded;
+        onEnded,
+        getMediaAttribute,
+        teardown;
     
-    document.addEventListener("impress:init", function(event) {
-        //root = event.target;
+    document.addEventListener("impress:init", function (event) {
+        root = event.target;
         api = event.detail.api;
         gc = api.lib.gc;
-        
+
         enhanceMedia();
         
-        gc.pushCallback(removeMediaClasses);
+        gc.pushCallback(teardown);
     }, false);
-
+    
+    teardown = function () {
+        var elements, el, i;
+        removeMediaClasses();
+        delete root.mediaAutoplay;
+        delete root.mediaAutopause;
+        delete root.mediaAutopstop;
+        elements = document.querySelectorAll("#root, .step, audio, video");
+        for (i = 0; i < elements.length; i += 1) {
+            el = elements[i];
+            delete el.mediaAutoplay;
+            delete el.mediaAutopause;
+            delete el.mediaAutostop;
+        }
+        for (i = 0; i < attributeTracker.length; i += 1) {
+            el = attributeTracker[i];
+            el.node.removeAttribute(el.attr);
+        }
+        attributeTracker = [];
+    };
+    
+    getMediaAttribute = function (attributeName, nodeList) {
```

Comment:
Could you avoid the name "nodeList". In the DOM there's a class "NodeList" and this is not of that type, so it would be better to use any other neutral name. Like just "nodes" or "nodeHierarchy".